interfaces/mount/backend.go

```diff
@@ -79,7 +80,19 @@ func (b *Backend) Setup(snapInfo *snap.Info, confinement interfaces.ConfinementO
 		return fmt.Errorf("cannot synchronize mount configuration files for snap %q: %s", snapName, err)
 	}
 	if err := UpdateSnapNamespace(snapName); err != nil {
-		return fmt.Errorf("cannot update mount namespace of snap %q: %s", snapName, err)
+		// try to discard the mount namespace and retry, but only if there aren't daemons in the snap
+		for _, app := range snapInfo.Apps {
+			if app.Daemon != "" {
+				return fmt.Errorf("Failed to update the mount namespace of snap %q when trying to update it, and cannot discard it and retry because it contains a daemon: %s", snapName, err)
```

Comment:
errors always start with lowercase and we avoid using failed but prefer "cannot ...", see CODING.md

Reply:
Ops... Fixed.